### PR TITLE
Some grammar fixes

### PR DESCRIPTION
**Pull request by jakozaur**

### CODE REVIEW

File: src/thesis.tex [hunk 1]
```diff
@@ -191,27 +191,27 @@
...
     The undisputed power of cloud computing constitutes
     in its elasticity and granularity: it allows users
-    to ask for hundreds of computers for only 5 minute usage which
-    are shipped during several minutes.
-    Such services are usually offered over remote network connection and users are billed
-    for the portion of the resources they have used.
+    to ask for hundreds of computers for only 5-minute usage which
+    are shipped for several minutes.
+    Such services are usually offered over a remote network connection and users are billed
+    for the portion of the resources, they have used.
```
Comment (mbalc) [inline src/thesis.tex:202]: I'm not sure about this comma

File: src/thesis.tex [hunk 2]
```diff
@@ -191,27 +191,27 @@
...
+    Nowadays, the industry increasingly relies on cloud technologies.
+    More and more companies start or move their products to a cloud environment
     for multiple reasons, including better scalability
-    and no need of architecture maintenance.
+    and no need for architecture maintenance.
     Unfortunately, this comes with additional financial costs imposed by cloud providers.
-    In order to make business profitable, companies try
-    to reduce amont of money they are supposed to pay to the minimum.
+    To make a business profitable, companies try
+    to reduce amount of money, they are supposed to pay to the minimum.
```
Comment (mbalc) [inline src/thesis.tex:214]: Same here

File: src/thesis.tex [hunk 3]
```diff
@@ -220,12 +220,12 @@
...
-    In spite of that fact, there is still place for new tools targeting
-    omitted types of clients or wrapping and bundling the greatest features from exisiting ones.
-    According to SumoLogic, the client requesting the software,
+    \textit{Cloudability}, \textit{Apptio}, \textit{CloudHealth} -- widely used nowadays.
+    In spite of that fact, there is still a place for new tools targeting
+    omitted types of clients or wrapping and bundling the most used features from existing ones.
+    According to Sumo Logic, the client requesting the software,
     development of which is the main topic of this thesis, there are methods of optimizing costs
-    that have a potential to save large amounts of money, while the technical details
+    that have the potential to save substential amounts of money, while the technical details
```
Comment (mbalc) [inline src/thesis.tex]: subst**a**ntial

File: src/thesis.tex [hunk 4]
```diff
@@ -236,7 +236,7 @@
     \section{Aim of the thesis}
 
     The primary objective of the thesis is to create a tool
-    complementing existing solutions used in cloud cost optimisation.
+    complementing existing solutions used in cloud cost optimization.
```
Comment (mbalc) [inline src/thesis.tex:239]: yeah right we should stick to american english

File: src/thesis.tex [hunk 5]
```diff
@@ -298,30 +298,30 @@
...
     However, according to our client, some businesses the client is in contact with
-    (as well as himself) reported a demand for a tool that would provide budgeting management
+    (as well as himself) reported demand for a tool that would provide budgeting management
     and basic machine termination automation.
     Such a tool may not cover as many use cases as the tools
     available on the market, but it would allow for huge savings while
     also being simple in implementation and maintenance.
-    The tooling that exists is targetting wideworld-scale companies
-    that are able to require expensive licenses and hire cost-optimization teams.
+    The tooling that exists is targetting wide world-scale companies
```
Comment (mbalc) [inline src/thesis.tex:311]: not sure about this

File: src/thesis.tex [hunk 6]
```diff
@@ -298,30 +298,30 @@
...
     The most important aspects of our tool are an intuitive interface and a variation
     of user notifications, which highly increase spendings clarity
     and help in decisions related to costs cutting.
     We hope it will allow companies to focus more on providing value to their clients
-    and have significantly lower costs in the same time without having to spend resources
+    and have significantly lower costs at the same time without having to spend resources
     on an extensive toolkit for cost optimization.
     This should be possible to achieve with budget and machine termination solutions
-    that are very simple in concept and implementation, but still provide great savings.
+    that are very simple in concept and implementation, but still, provide significant savings.
```
Comment (mbalc) [inline src/thesis.tex:324]: same thing about this comma

File: src/thesis.tex [hunk 7]
```diff
@@ -350,21 +350,21 @@
     \end{enumerate}
 
     Some of the most prominent tools currently available
-    on the market that improve resource management experience
+    on the market that improves resource management experience
```
Comment (mbalc) [inline src/thesis.tex]: imo it should be `improve` - the tools improve, they're plural

File: src/thesis.tex [hunk 8]
```diff
@@ -465,20 +465,20 @@
...
     Most commonly used options are either their public APIs~\cite{AWSCostManagement}
     or specific SDKs supporting lots of languages.
-    Unfortuantely those tools are rather simple and do not satisfy
+    Unfortunately, those tools are rather simple and do not satisfy
     all of the needs of potential clients.
 
     \chapter{Tool for AWS cost optimization}
 
-    In this chapter we will describe the whole application -- \textit{Omigost}.
+    In this chapter, we will describe the entire application -- \textit{Omigost}.
```
Comment (mbalc) [inline src/thesis.tex:481]: i have to check up this comma

File: src/thesis.tex [hunk 9]
```diff
@@ -489,11 +489,11 @@
     \end{enumerate}
 
     as well as uses Slack integration as a convenient and not overly intrusive mean
-    of providing employees a simplified way of saving company's money in their day-to-day work.
+    of providing employees a simplified way of saving a company's money in their day-to-day work.
```
Comment (mbalc) [inline src/thesis.tex:492]: imo money isn't a countable noun, therefore `a` shouldn't be here

File: src/thesis.tex (hunk 8 above)
Comment (mbalc) [reply src/thesis.tex:481]: cause imo it's optional, the choice slightly changes meaning and the previous one felt more right for me - whereas the new one makes it less straightforward with the intent

File: src/thesis.tex [hunk 10]
```diff
@@ -572,28 +572,28 @@
       \caption{\label{fig:arch-diag} Draft diagram of the tool architecture}
     \end{figure}
 
-    The architecture of our tool can be described as a traditional client-server architecture.
+    The architecture of our tool can be described as traditional client-server architecture.
```
Comment (mbalc) [inline src/thesis.tex:575]: not sure here

File: src/thesis.tex [hunk 11]
```diff
@@ -572,28 +572,28 @@
...
     using the frontend client application
-    served by the Omigost server or, alternatively, from any kind of a storage
+    served by the Omigost server or from any storage
     service like, for example, Amazon's S3.
     Those requests are then handled by a backend application running on a central server machine.
-    During the handling process our application uses both a local instance
+    During the handling process, our application uses both a local instance
     of a PostgreSQL relational database
-    and a connection with selected Amazon Web Services via a SDK library.
+    and a connection with selected Amazon Web Services via an SDK library.
```
Comment (mbalc) [inline src/thesis.tex:583]: checked this up, wow didn't know about this

File: src/thesis.tex [hunk 12]
```diff
@@ -634,7 +634,7 @@
     Its main roles are to:
     \begin{itemize}
         \item request and parse raw data from backend into visual representations of it,
-        \item provide an interface for the user,
+        \item provides an interface for the user,
```
Comment (mbalc) [inline src/thesis.tex:637]: imo should be `provide` - we list responsibilities, so it is like if each `item` was prepended with `Its main roles are to`

File: src/thesis.tex [hunk 13]
```diff
@@ -647,8 +647,8 @@
     using the React~\cite{React} framework.
     Typescript offers both great flexibility and type safety compared to vanilla Javascript.
     It is the most common alternative to Javascript,
-    supported by large bussiness insitutions, like Microsoft.
-    We decided that other alternatives,
+    supported by large business institutions, like Microsoft.
+    We decided those other alternatives,
```
Comment (mbalc) [inline src/thesis.tex]: not sure about this

File: src/thesis.tex [hunk 14]
```diff
@@ -677,26 +677,26 @@
...
+    The Redux store is central storage for the information
     about current module settings but also serves view routing data,
     state of dialogs and notifications.
 
     We also provided \textit{Jest} test suites
     as well as a documentation generated by \textit{tsdoc}.
     We also decided to utilize \textit{Storybook}
     to easily design and visually test frontend components,
-    which for users stands for a complement
+    which for users stands for a compliment
```
Comment (mbalc) [inline src/thesis.tex]: no, here it should be `complement` imo

File: src/thesis.tex [hunk 15]
```diff
@@ -706,14 +706,14 @@
     It is easy to use with various IDEs and has a flexible configuration.
     The main competitor of Gradle is Maven so we have considered both.
     For us, Gradle turned out to be the better choice because
-    of its performance and usage convenience.
+    of its performance and user convenience.
 
-    For data persistence we have chosen a PostgreSQL database.
+    For data persistence, we have chosen a PostgreSQL database.
```
Comment (mbalc) [inline src/thesis.tex:711]: same "clarity" argument as in "In this chapter, ..."

File: src/thesis.tex [hunk 16]
```diff
@@ -731,7 +731,7 @@
 
     Following a straightforward instruction, everyone is able
     to create their own Docker image of the application.
-    Moreover, all of the configuration is present in one file,
+    Moreover, all of the configurations are present in one file,
```
Comment (mbalc) [inline src/thesis.tex]: imo we rather mean a single configuration here but not sure

File: src/thesis.tex [hunk 17]
```diff
@@ -790,11 +790,11 @@
     
     \includegraphics[width=\textwidth] {imgs/screenshots/screen_accounts_browse.png}
     
-    Similar view exists in the users tab to present all created user accounts in the system.
+    The similar view exists in the user's tab to present all created user accounts in the system.
```
Comment (mbalc) [inline src/thesis.tex]: it's a tab about users, not a tab owned by a user

File: src/thesis.tex [hunk 18]
```diff
@@ -1032,10 +1032,10 @@
     \section{Organization of work}
 
     We have been working in iterations that lasted around one month.
-    At the beginning we defined and created new issues.
-    During the iteration everyone chose desirable task,
+    In the beginning, we defined and created new issues.
```
Comment (mbalc) [inline src/thesis.tex:1035]: not sure about all these commas here too

**Review by mbalc — CHANGES_REQUESTED** [review]
in a few places I'm certain that the change was wrong; in a few others I marked them for myself and for you to double check on them - I'd have to do a more thorough research about these comma use cases but I don't have time for this at the moment

**Comment by mbalc** [comment]
@jakozaur I marked as resolved all but those issues that certainly need a fix or response in my opinion - I can take care of them today evening but if you have some time to check out these thoughts and maybe add some comments on them then I'd appreciate that

**Comment by jakozaur** [comment]
I can have a look at today sometime between 4 pm to 6 pm.

File: src/thesis.tex (hunk 3 above)
Comment (jakozaur) [reply src/thesis.tex]: fixed

File: src/thesis.tex (hunk 14 above)
Comment (jakozaur) [reply src/thesis.tex]: fixed

File: src/thesis.tex (hunk 16 above)
Comment (jakozaur) [reply src/thesis.tex]: fixed

File: src/thesis.tex (hunk 17 above)
Comment (jakozaur) [reply src/thesis.tex]: fixed

File: src/thesis.tex (hunk 9 above)
Comment (mbalc) [reply src/thesis.tex:492]: umm my fault, the `a` refers to `company`, not `money`

**Pull request merged**